<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SetupMidi</class>
 <widget class="QDialog" name="SetupMidi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>480</width>
    <height>865</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>MIDI Setup</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="0">
    <widget class="QGroupBox" name="timidity_extra">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Timidity Setup</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="timidityCfgPathLabel">
        <property name="text">
         <string>Config file path:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="timidityCfgPath"/>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="timidityCfgPathBrowse">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QGroupBox" name="opnmidi_extra">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>OPNMIDI Extra Setup</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4" columnstretch="0,1000">
      <item row="1" column="1">
       <widget class="QComboBox" name="opnVolumeModel">
        <item>
         <property name="text">
          <string>[Auto]</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Generic</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Native OPN2</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>DMX</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Apogee</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>9X</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="opnEmulator">
        <item>
         <property name="text">
          <string>Mame YM2612 (OPN2)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Nuked OPN2 (YM3438)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>GENS/GS II OPN2</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>YMFM OPN2</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Neko Project II OPNA</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Mame YM2608 (OPNA)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>YMFM OPNA</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Nuked OPN2 (YM2612)</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="5" column="0" colspan="2">
       <widget class="QCheckBox" name="opn_autoArpeggio">
        <property name="text">
         <string>Enable automatic arpeggio</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="opnNumChips">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>8</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Volume ranges model:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QFrame" name="frame">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>24</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QCheckBox" name="opn_use_custom">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Check this to use custom bank.&lt;/p&gt;&lt;p&gt;Uncheck this if you want to use default bank and keep custom bank field be filled.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="styleSheet">
            <string notr="true">QCheckBox {
	border-color: rgb(0, 170, 0);
    border-width: 1px;
	color: rgb(0, 255, 255);
 }</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>Custom bank file:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="opn_bank"/>
         </item>
         <item>
          <widget class="QPushButton" name="opn_bank_browse">
           <property name="text">
            <string>Browse</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="adlEmulatorLabel_2">
        <property name="text">
         <string>Chip emulator:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Number of chips:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Channel alloc mode:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="opnChanAlloc">
        <item>
         <property name="text">
          <string>[Auto]</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sounding off delay based</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Same instrument</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Any first released</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="6" column="0">
    <widget class="QLineEdit" name="autoArguments">
     <property name="statusTip">
      <string>The list of music path arguments that you can use to reuse this setup.</string>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QGroupBox" name="adlmidi_extra">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>ADLMIDI Extra Setup</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="1" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>ADL Midi bank ID:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="adl_bankId">
        <property name="currentIndex">
         <number>-1</number>
        </property>
        <property name="maxVisibleItems">
         <number>50</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="adlEmulatorLabel">
        <property name="text">
         <string>Chip emulator:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="adlEmulator">
        <item>
         <property name="text">
          <string>Nuked OPL3 1.8</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Nuked OPL3 1.7</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>DosBox OPL3</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Opal OPL3 (by RAD)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>JavaOPL3 (OPL3.java)</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Volume ranges model:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QComboBox" name="adlVolumeModel">
        <item>
         <property name="text">
          <string>[Auto]</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Generic</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Native OPL3</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">DMX</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">Apogee Sound System</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">9X (SB16)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>DMX (Fixed AM)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Apogee (Fixed AM)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">Audio Interfaces Library</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">9X (Generic FM)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string notr="true">HMI Sound Operating System</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>HMI Sound Operating System (Old)</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="7" column="0" colspan="2">
       <widget class="QFrame" name="frame_3">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="spacing">
          <number>5</number>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QCheckBox" name="adl_use_custom">
           <property name="toolTip">
            <string>Check this to use custom bank.
Uncheck this if you want to use default bank and keep custom bank field be filled.</string>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Custom bank file:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="adl_bank"/>
         </item>
         <item>
          <widget class="QPushButton" name="adl_bank_browse">
           <property name="text">
            <string>Browse</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="8" column="0" colspan="2">
       <widget class="QCheckBox" name="adl_tremolo">
        <property name="text">
         <string>Tremolo</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <property name="tristate">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="9" column="0" colspan="2">
       <widget class="QCheckBox" name="adl_vibrato">
        <property name="text">
         <string>Vibrato</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <property name="tristate">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="10" column="0" colspan="2">
       <widget class="QCheckBox" name="adl_scalableModulation">
        <property name="text">
         <string>Scalable modulation</string>
        </property>
        <property name="tristate">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="12" column="0" colspan="2">
       <widget class="QPushButton" name="resetDefaultADLMIDI">
        <property name="text">
         <string>Reset to defaults</string>
        </property>
       </widget>
      </item>
      <item row="11" column="0" colspan="2">
       <widget class="QCheckBox" name="adl_autoArpeggio">
        <property name="text">
         <string>Enable automatic arpeggio</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Number of chips:</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QSpinBox" name="adlNumChips">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
        <property name="value">
         <number>4</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Channel alloc mode:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QComboBox" name="adlChanAlloc">
        <item>
         <property name="text">
          <string>[Auto]</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sounding off delay based</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Same instrument</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Any first released</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QGroupBox" name="midi_setup">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>MIDI Setup</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>MIDI Synthesizer:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="mididevice">
        <item>
         <property name="text">
          <string>ADL Midi (OPL3 Synth emulation)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Native midi (Built-in MIDI of your OS)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Timidity (GUS-like wavetable MIDI Synth)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>OPN Midi (YM2612 Synth emulation)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>FluidSynth (SF2-based wavetable MIDI Synth)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Emu De Midi (OPLL, PSG, SCC synth emulation)</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QLineEdit" name="midiRawArgs">
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Accepts list of letter-marked integer arguments separated by semicolons. &lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;Example: &lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-family:'monospace'; text-decoration: underline;&quot;&gt;s0;b62;t0;v0;a0;m0;&lt;/span&gt;&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;All arguments are optional. &lt;span style=&quot; font-style:italic;&quot;&gt;Last argument must be closed with semicolon&lt;/span&gt;. &lt;/p&gt;&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 1;&quot;&gt;&lt;li style=&quot; margin-top:12px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;s - Synthesizer type: &lt;/li&gt;&lt;ul style=&quot;margin-top: 0px; margin-bottom: 0px; margin-left: 0px; margin-right: 0px; -qt-list-indent: 2;&quot;&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;0 - ADLMIDI, emulated &lt;a href=&quot;https://wohlsoft.ru/pgewiki/FM_Synthesis&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;OPL3 (YMF262) synthesizer&lt;/span&gt;&lt;/a&gt; with loop tags support. Working by Default. (Recorded examples to listen are presented &lt;a href=&quot;https://wohlsoft.ru/pgewiki/FM_Synthesis&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;here&lt;/span&gt;&lt;/a&gt;). OPL3 chip was widely used on PC sound cards in 80/90'th years.&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;1 - Native MIDI, uses default operating system MIDI interface. Not supports loop tags and system exclusive messages.*** &lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;2 - Timidity, requires patches bank in the &amp;quot;timidity&amp;quot; folder in the application path (can be customized with &amp;quot;MIX_Timidity_addToPathList(const char*path)&amp;quot; function called before library initialization) **&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;3 - OPNMIDI, emulated &lt;a href=&quot;https://wohlsoft.ru/pgewiki/FM_Synthesis&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;OPN2 (YM2612) synthesizer&lt;/span&gt;&lt;/a&gt; with loop tags support. OPN2 chip is used on Sega MegaDrive / Genesis game console. This synthesizer has been added &lt;a href=&quot;https://github.com/WohlSoft/PGE-Project/commit/d7f8868a80409ba05f7df854f0fb5011d50632ec&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#0000ff;&quot;&gt;since May 14, 2017&lt;/span&gt;&lt;/a&gt;.&lt;/li&gt;&lt;/ul&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;b - (ADLMIDI) Set bank ID (look for available banks list into PGE MusPlay application to preview them). Default is 58.&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;t - (ADLMIDI) Deep tremolo (1 or 0), increase amplitude of tremolo effect for instruments which are using tremolo OPL flag. Default 1.&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;v - (ADLMIDI) Deep vibrato (1 or 0), increase amplitude of vibrato effect for instruments which are using vibrato OPL flag. Default 1.&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;a - (ADLMIDI) Enables AdLIB drums mode (1 or 0), enables legacy OPL2 drums mode. Default 0.&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;m - (ADLMIDI) Enables scalable modulation (1 or 0), Allows dynamic scaling of the modulator result with volume. Default 0.&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;c - (ADLMIDI and OPNMIDI) Count of chip emulators are allowing to excite limit of voice channels. Default 4. Allowed value from 1 to 100.&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;f - (ADLMIDI) Count of four-operator channels between of all chip emulators (6 channels maximum per every chip. For example, you have using 2 chips, you can have 12 four-op channels maximum).&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;l - (ADLMIDI and OPNMIDI) Volume scaling model code (0 - auto, 1 - Generic, - 2 OPL3-Native, 3 - DMX, 4 - Apogee, - 5 - Win9X-like) - an algorithm to scale a volume level, affects music expressionism.&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;r - (ADLMIDI and OPNMIDI) Use full-ranged CC74 Brightness MIDI controller (1 or 0). Default 0.&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;e - (ADLMIDI) Choose OPL3 emulator: 0 - Nuked OPL3 (Very accurate, may lag on older CPUs), 1 - Nuked OPL3 1.7.4 that was optimized by Troosh, and 2 - DosBox 0.74 (Well-accurate, fastest).&lt;/li&gt;&lt;li style=&quot; margin-top:0px; margin-bottom:12px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;e - (OPNMIDI) Choose OPN2 emulator: 0 - Mame YM2612 (Well-accurate, fast), 1 - Nuked OPN2 (Very accurate, requires VERY powerful CPU. Suggested to use up to 2 chips max with it), 2 - GENS 2.10 (Very outdated, inaccurate, but fastest)&lt;/li&gt;&lt;/ul&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="placeholderText">
         <string>Type raw MIDI arguments here or leave empty to use GUI setup...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QGroupBox" name="fluidsynth_extra">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="acceptDrops">
      <bool>false</bool>
     </property>
     <property name="title">
      <string>FluidSynth Setup</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>SoundFonts paths:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="fluidSynthSF2Paths"/>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="fluidSynthSF2PathsBrowse">
        <property name="text">
         <string>Browse...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="5" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Maximum</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
